xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -457,23 +402,26 @@ void CPeripheralBusAddon::UpdateAddons(void)
   {
     CLog::Log(LOGDEBUG, "Add-on bus: Registering add-on %s", addonId.c_str());
 
-    auto GetAddon = [&addonId](const PeripheralAddonPtr& addon) { return addon->ID() == addonId; };
+    auto GetAddon = [&addonId](const BinaryAddonBasePtr& addon) { return addon->ID() == addonId; };
 
-    PeripheralAddonVector::iterator it = std::find_if(newAddons.begin(), newAddons.end(), GetAddon);
+    BinaryAddonBaseList::iterator it = std::find_if(newAddons.begin(), newAddons.end(), GetAddon);
     if (it != newAddons.end())
     {
-      PeripheralAddonPtr& newAddon = *it;
-
-      bool bCreated;
+      PeripheralAddonPtr newAddon = std::make_shared<CPeripheralAddon>(*it);
+      if (newAddon)
       {
-        CSingleExit exit(m_critSection);
-        bCreated = (newAddon->CreateAddon() == ADDON_STATUS_OK);
-      }
+        bool bCreated;
 
-      if (bCreated)
-        m_addons.emplace_back(std::move(newAddon));
-      else
-        m_failedAddons.emplace_back(std::move(newAddon));
+        {
+          CSingleExit exit(m_critSection);
+          bCreated = newAddon->CreateAddon();
+        }
+fprintf(stderr, "-----3- %s - %i\n", __PRETTY_FUNCTION__, bCreated);
```

Comment:
No worries. last week i sent our customers code which spewed 'hore' five times on every run of the simulator :) (add a w in front to translate the norwegian..) proud moment ;)